src/App.js

```diff
@@ -1,12 +1,24 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
+import calculate from './logic/calculate';
 import Calculator from './components/Calculator';
 
-export default class App extends Component {
-  render() {
-    return (
-      <>
-        <Calculator />
-      </>
-    );
-  }
-}
+const App = () => {
+  const [state, setState] = useState({
+    total: 0,
+    next: 0,
+  });
+
+  const clickHandler = (e) => {
+    setState((prev) => calculate(prev, e.target.value));
+  };
+
+  const { total, next } = state;
+  const stateOfCalc = { total, next };
+  return (
+    <>
+      <Calculator handleClick={clickHandler} calcState={stateOfCalc} />
```

Comment:
[OPTIONAL] I think you don't need to deconstruct the object here, to construct a new object to pass on props. You could just pass the state in `calcState={state}` and save a couple of lines. :sweat_smile: